### PR TITLE
Add bitbucket support to finding view SCM urls

### PR DESCRIPTION
**Pull request by drJabber**
**Description**
Adds support of bitbucket source code navigation URLs to SCM URLs in Finding views

**Test results**
Added tests in test_finding_model.py

**Documentation**
Documentation updated in  docs/content/en/integrations/source-code-repositories.md

### CODE REVIEW

**Comment by drJabber** [comment]
Hello, Team, could somebody approve PR? Although it modifies models.py it does not modify DB model itself - it just modifies Finding method, that composes SCM url to navigate to source code in browser, so I think it could be included into code base before v3 release.

File: dojo/settings/settings.dist.py [hunk 1]
```diff
@@ -1468,7 +1468,7 @@ def saml2_attrib_map_format(dict):
     'kube-bench Scan': DEDUPE_ALGO_HASH_CODE,
     'Threagile risks report': DEDUPE_ALGO_UNIQUE_ID_FROM_TOOL_OR_HASH_CODE,
     'Humble Json Importer': DEDUPE_ALGO_HASH_CODE,
-    'MSDefender API': DEDUPE_ALGO_HASH_CODE,
+    'MSDefender Parser': DEDUPE_ALGO_HASH_CODE,
```
Comment (manuel-sommer) [inline dojo/settings/settings.dist.py:1471]: There is something wrong with your PR. Please fix your commit history.

**Review by manuel-sommer — CHANGES_REQUESTED** [review]

File: dojo/settings/settings.dist.py (hunk 1 above)
Comment (Maffooch) [reply dojo/settings/settings.dist.py:1471]: +1 ^

Comment (drJabber) [reply dojo/settings/settings.dist.py:1476]: fixed, thank you

Comment (manuel-sommer) [reply dojo/settings/settings.dist.py:1476]: No, it is not fixed.

Comment (drJabber) [reply dojo/settings/settings.dist.py:1476]: sorry, @manuel-sommer , what should i do to fix commit history? I thought I just should rebase to DefectDojo:dev ...
or may be should i squash my several commits into single?

Comment (manuel-sommer) [reply dojo/settings/settings.dist.py:1476]: This link will probably help: 
https://github.com/openedx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

Comment (drJabber) [reply dojo/settings/settings.dist.py]: tried to rebase as described in this manual, I hope now it's ok

**Review by Maffooch — APPROVED** [review]
Great job "feature flagging" this new behavior with the product meta

**Review by mtesauro — APPROVED** [review]
Approved

File: docs/content/en/integrations/source-code-repositories.md [hunk 2]
```diff
@@ -7,18 +7,47 @@ weight: 5
...
+- for standalone/onpremise BitBucket https://bb.example.com/scm/some-project/some-repo.git or https://bb.example.com/scm/some-user-name/some-repo.git for user public repo (like git clone url)
+![Edit Engagement (Bitbucket standalone)](../../../static/images/source-code-repositories-bitbucket-onpremise_1.png)
+
+For CI/CD Engagement, where user could set commit hash, branch/tag and code line it should look like examples below:
+- for GitHub - like https://github.com/DefectDojo/django-DefectDojo 
+- for GitLab - like https://gitlab.com/gitlab-org/gitlab
+- for public BitBucket - like https://bitbucket.org/some-user/some-project.git (like git clone url)
+- for standalone/onpremise BitBucket https://bb.example.com/scm/some-project.git or https://bb.example.com/scm/some-user-name/some-repo.git for user public repo (like git clone url)
+
+If user does not set commit hash or branch/tag in appropriate fields of CI/CD Engagement edit form - the URL should be look like in Interactive Engagement edit form.
```
Comment (cneill) [inline docs/content/en/integrations/source-code-repositories.md:30]:
```suggestion
If user does not set commit hash or branch/tag in appropriate fields of CI/CD Engagement edit form, the URL should look like in Interactive Engagement edit form.
```

File: docs/content/en/integrations/source-code-repositories.md [hunk 3]
```diff
@@ -7,18 +7,47 @@ weight: 5
...
+
+SCM navigation URL is composed from Repo URL using SCM Type. Github/Gitlab SCM type is default, but user could set certain SCM type in Product custom field "scm-type".s
```
Comment (cneill) [inline docs/content/en/integrations/source-code-repositories.md:32]:
```suggestion
SCM navigation URL is composed from Repo URL using SCM Type. Github/Gitlab SCM type is default, but user could set certain SCM type in Product custom field "scm-type".
```

File: dojo/models.py [hunk 4]
```diff
@@ -3027,26 +3027,125 @@ def get_file_path_with_link(self):
...
+                if st:
+                    return st.lower()
+        return 'github'
+
+    def bitbucket_public_prepare_scm_base_link(self, uri):
+        # bitbucket public (https://bitbucket.org) url template for browse is:
+        # https://bitbucket.org/<username>/<repository-slug>
+        # but when you get repo url for git, its template is:
+        # https://bitbucket.org/<username>/<repository-slug>.git
+        # so to create borwser url - git url should be recomposed like below:
```
Comment (cneill) [inline dojo/models.py:3046]:
```suggestion
        # so to create browser url - git url should be recomposed like below:
```

File: dojo/models.py [hunk 5]
```diff
@@ -3027,26 +3027,125 @@ def get_file_path_with_link(self):
...
+
+        parts_uri = uri.split('.git')
+        return parts_uri[0]
+
+    def bitbucket_public_prepare_scm_link(self, uri):
+        # if commit hash or branch/tag is set for engagement/test -
+        # hash or barnch/tag should be appended to base browser link
```
Comment (cneill) [inline dojo/models.py:3053]:
```suggestion
        # hash or branch/tag should be appended to base browser link
```

**Review by cneill — CHANGES_REQUESTED** [review]
Just a few small things, otherwise looks good!

**Comment by drJabber** [comment]
> Just a few small things, otherwise looks good!

thanks, commited

**Review by manuel-sommer — APPROVED** [review]

**Review by cneill — APPROVED** [review]

**Comment by drJabber** [comment]
still need one more reviewer approval. Could anybody with write access review this, please?

**Comment by drJabber** [comment]
Hi @grendel513 / @devGregA / @blakeaowens, I'd like to ask for a review

**Review by blakeaowens — APPROVED** [review]

**Pull request merged**